app/models/deploy_service.rb

```diff
@@ -5,11 +5,12 @@ def initialize(user)
     @user = user
   end
 
-  def deploy!(stage, reference)
-    deploy = stage.create_deploy(reference: reference, user: user)
+  def deploy!(stage, reference = nil, attributes = {})
```

Comment:
that looks wrong ... I was expecting `deploy!(stage, attributes)` since both are mandatory